src/manifest.c

```diff
@@ -503,17 +495,19 @@ static int retrieve_manifests(int current, int version, char *component, struct
 
 	string_or_die(&filename, "%s/%i/Manifest.%s.tar", state_dir, version, component);
 	if (stat(filename, &sb) == 0) {
-		return 0;
+		ret = 0;
+		goto out;
```

Comment:
I'm running into a segfault with this change. The "goto out" will lead to freeing "url", which is not initialized at this point.